lib/workers/repository/process/write.ts

```diff
@@ -35,7 +42,20 @@ export async function writeUpdates(
   for (const branch of branches) {
     addMeta({ branch: branch.branchName });
     const branchExisted = branchExists(branch.branchName);
-    const res = await processBranch(branch);
+    const branchCache =
+      cachedBranches?.find((br) => br.branchName === branch.branchName) ??
+      ({} as BranchCache);
+
+    const managersHash = hasha(
+      branch.upgrades
+        .map((upgrade) => hashMap.get(upgrade.manager) ?? upgrade.manager)
+        .filter(is.string)
+    );
+    branch.configAndManagersHash = hasha([
+      JSON.stringify(config),
```

Comment:
@rarkins should we really use the big config object here? Isn't this a litle bit late? Should we better split or move to an earlier stage?

🤔 🤔 🤔 🤔 

we already store a `configHash`

https://github.com/renovatebot/renovate/blob/bee604a1a3b9c8da099d2908a2e68263fa831eea/lib/workers/repository/process/extract-update.ts#L78

Reply:
`configHash` won't contain fields related to updates right like `upgrades`, which is want we want to verify with this check.

Reply:
The existing `configHash` is meant to be narrowed to only fields which impact the extract logic. For here we need it to also consider the branch/upgrades logic.

This "big" config is definitely too big, but it's a start. Making it too big will mean that it invalidates itself too often, but that's not a *problem*.

Yes to renaming the hashes to make it more understandable

Reply:
How about we can call it branchConfigHash for now and in the next PR I will dump this one and create 2 different ones with names branchUpgradeConfigHash and branchPrConfigHash.

Reply:
I think there's some advantage to using `fingerprint` terminology to indicate its purpose